src/main/java/seedu/algobase/ui/MainWindow.java

```diff
@@ -238,4 +241,32 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    /**
+     * Executes a UI action returns the result.
+     *
+     * @see seedu.algobase.ui.action.UiLogic#execute(UiActionDetails)
+     */
+    private UiActionResult executeUiAction(UiActionDetails uiActionDetails) {
+        try {
+            UiActionResult uiActionResult = uiLogic.execute(uiActionDetails);
+            uiActionResult.getFeedbackToUser().ifPresent((feedback) -> {
+                logger.info("Result: " + feedback);
+                resultDisplay.setFeedbackToUser(feedback);
+            });
+
+            if (uiActionResult.isShowHelp()) {
+                handleHelp();
+            }
+
+            if (uiActionResult.isExit()) {
+                handleExit();
+            }
+
+            return uiActionResult;
+        } catch (UiActionException | ParseException e) {
+            resultDisplay.setFeedbackToUser(e.getMessage());
```

Comment:
Probably it's better to display UI results at the bottom of the app (so that `resultDisplay` always shows feedback from CLI commands`

Reply:
I agree that a snackbar would be useful, but I think it is fine if significant changes are shown to the user, e.g. editing and deleting problems, as compared to switching and opening tabs.